### PR TITLE
Api error messages

### LINKED ISSUE #3322: Add custom error message with translation for existing unique validations in the schema.graphql file
**Problem** 

As a user I want to see  user friendly error messages translated to my preferred language (EN / FR ) 
for all the unique values in the system ( to start with )  

**A/C** 

- set up validation errors
- Create validation rule with error code
- Suggest conventions for unique, understandable error codes
- Create a frontend class to collect localized error messages keyed to error codes
- Create a helper function to take a error code and turn it into a localized error message for toasting
- Unit test with English and French validation

    
**Implementation Details** 
- [ ] Add custom error message in graphQL createUserInput mutation for 
- sub and  email field in User table,
-  key in skill_families table

- [ ] Add error message in common messages for translation 
- [ ] Use the added message that has been added in  common messages inside the API handler to bubble up the error on the front end

### PR DESCRIPTION
**Pull request by petertgiles**
This branch adds localized API error messages.  Validation rules were added to the schema with error messages.  An error message class with intl messages was added to the common project.  A helper function was added to find the message with the matching string.  A convention was suggested in the comments to use the source rule name as the key in the error message object.

## Notes
- A test was added for the function to extract the error messages.  I couldn't reliably test the toasts themselves in Jest though.  I'm not sure that would be super useful anyway.
- The duplicate `ClientProvider` in talentsearch was removed.  I couldn't easily remove the one in IAP so I posted #3668.

## Limitations
-  In this implementation the messages must be manually added to a common file.  Someday it would be nice to have our intl utilities scan and find these strings automatically.  #3330
-  The static messages from the validators was added as well but not the parametric ones.  Building the parameter on the fly would not be useful since the substituted field names would not be translated.  It would be possible to add strings for each field if that is found to be useful but most validation will be happening on the front end anyway.

## Testing
- [ ] Add these strings to your common fr.json file and rerun `npm run intl-compile` and `npm run dev`
```
"l8p1el": {
    "defaultMessage": "FR This user identifier (sub) is already in use FR",
    "description": "Error message that the given user identifier is already in use."
  },
"nDaEgr": {
    "defaultMessage": "FR This email address is already in use FR",
    "description": "Error message that the given email address is already in use."
  },
```
- [ ] Trigger validation errors by attempting to add a new user with existing email and sub values (like "applicant@test.com") in the admin site
- [ ] Observe the errors in English and French
image

Closes #3322

### CODE REVIEW

**Comment by vd1992** [comment]
- added to the intl file in `\common\src\lang\fr.json`
- ran intl-compile and dev 

if I only add in a duplicate email, no duplicate sub this happens

image

if it is a duplicate email _and_ duplicate sub, it works

image

**Comment by vd1992** [comment]
checking elsewhere, it works! but could probably be improved

this one outputs the same message twice, presumably occurs twice due to location_en AND location_fr

image

**Comment by vd1992** [comment]
trying with no essential skills attached

image

no essential skills attached as well as missing locations 

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
great work Peter! got the ball rolling on a very handy feature
messages will take time to build up and sort out, but good start

seems like it kinda glitches up when there is only one message to pass in, that probably ought to be fixed lol

File: frontend/common/src/components/ClientProvider/ClientProvider.tsx [hunk 1]
```diff
@@ -81,10 +86,75 @@ const willAuthError = ({ authState }: { authState: AuthState | null }) => {
...
+        {localizedMessages.map((message) => (
+          <li key={message}>{message}</li>
+        ))}
+      </ul>
+    );
+  }
+
+  // if just 1, toast by itself
+  if (localizedMessages.length === 1) {
+    return <span>localizedMessages[0]</span>;
```
Comment (vd1992) [inline frontend/common/src/components/ClientProvider/ClientProvider.tsx:146]: I think this is the cause
should be in `{}`

Comment (petertgiles) [reply frontend/common/src/components/ClientProvider/ClientProvider.tsx]: Doh, how did I miss that?  Good catch!

**Review by vd1992 — APPROVED** [review]
my environment is busy with hydrogen, so I will just have faith in you 👍

**Pull request merged**